Best practices/React/Organization.md

```diff
@@ -4,11 +4,111 @@
 
 This section describes a sensible starting point for organizing a React application that supports server-side rendering.
 
-## `/app`
+## Application
 
-### `/app/components` and `<Component>/components`
+### React Components and Sub-components
+
+React Components should always exist in a `components/` folder, but the component's internal structure may change depending on their complexity.
```

Comment:
Could we define the difference between the three, `foundation`,`features`, and `components`?

Reply:
`root/foundation`: the base of the application that is usually render once. Such as Providers of the application, routing setup, and frame (sidebar, topbar) of the application. It typically contains react component that is not meant for reuse anywhere else.

`root/features`: different parts of the application, typically corresponded to top level url. Majority of the application lives here.

`root/components`: top level components works much like the rest of the `components`. It contains React components meant to be share and re-use by any file in the same folder or below.

Reply:
I feel like that might be project level config and I'm trying to scope this exclusively to components,  but that's a great point. Are `root/features` similar to `sections`? I've actually never heard these conventions before, are they unique to `web`, or do we use them elsewhere as well?

Maybe I should just remove this sentence? or change `always` to usually and list the exceptions you mentioned?

Reply:
`root/features` = `sections`, just the future version :)

I do agree it may be out of scope. Deleting the word `always` and adding the `root/components` explanation in here to the extend it make sense should be enough.

> React Components can often be found in `components` folder. These folder are meant to house share and re-useable React Components that should only be consume by other components in the same folder or below.